tests/output/ns-nsplex.out

```diff
@@ -1,3 +1,6 @@
+Degree of FEM Space: 1
 Global FEM nodes: 0 on 1 ranks
 Local FEM nodes: 8 (0 owned)
+dm_plex_box_faces: NONE
+Time taken for solution: 0.00363827
```

Comment:
Any way to have this accept any number value? Like a regex:

```suggestion
Time taken for solution: (?:\d?\.\d+)
```

Reply:
You'd have to rework lines 130-147 of `tests/tap.sh` to make that work. 

What I'd probably do is check that the desire value is within tolerances in the code and print some sort of good/bad message in test mode for that line. Though, does wallclock time help us check something important here?

Reply:
In `bps.c`, we have a command line arg for testing mode, and we modify/reduce output if we are in testing mode.

Example:
```
  // Print summary
  if (!test_mode) {
    PetscInt P = degree + 1, Q = P + qextra;
    const char *usedresource;
    CeedGetResource(ceed, &usedresource);
    ierr = PetscPrintf(comm,
                       "\n-- CEED Benchmark Problem %d -- libCEED + PETSc --\n"
                       "  libCEED:\n"
                       "    libCEED Backend                    : %s\n"
                       "  Mesh:\n"
                       "    Number of 1D Basis Nodes (p)       : %d\n"
                       "    Number of 1D Quadrature Points (q) : %d\n"
                       "    Global nodes                       : %D\n"
                       "    Owned nodes                        : %D\n",
                       bpChoice+1, usedresource, P, Q, gsize/ncompu,
                       lsize/ncompu); CHKERRQ(ierr);
  }
```

Reply:
The wall clock time is just a measurement, so we could leave it off completely for testing if I setup a separate testing mode.

Reply:
I'm personally a big fan of 'pretty' output normally and stripped down output in testing mode. I generally only print in test mode on an error.

I've mentioned `bps.c` a few times in this review because this Navier-Stokes code started as a modified version of that file. We have since improved that example so there are fixes in there for some of the things you're hitting.